tools/linker/MonoTouch.Tuner/MonoTouchTypeMap.cs

```diff
@@ -76,41 +72,57 @@ bool IsNSObject (TypeDefinition type)
 			cached_isnsobject.Add (type);
 			return true;
 		}
-		
-		// type has a "public .ctor (IntPtr)" with a [CompilerGenerated] attribute
-		static bool IsGeneratedBindings (TypeDefinition type, DerivedLinkContext link_context)
+
+		bool IsWrapperType (TypeDefinition type)
 		{
-			if (type.IsNested)
-				return IsGeneratedBindings (type.DeclaringType, link_context);
-			
-			if (!type.HasMethods)
+			var registerAttribute = LinkContext.StaticRegistrar.GetRegisterAttribute (type);
+			return registerAttribute?.IsWrapper == true || registerAttribute?.SkipRegistration == true;
+		}
+
+		bool IsCIFilter (TypeReference type)
+		{
+			if (type == null)
 				return false;
-			
-			foreach (MethodDefinition m in type.Methods) {
-				if (!m.IsConstructor)
-					continue;
-				if (!m.HasParameters)
-					continue;
-				if (m.Parameters.Count != 1)
-					continue;
-				if (!m.Parameters [0].ParameterType.Is ("System", "IntPtr"))
-					continue;
-				return m.IsGeneratedCode (link_context);
-			}
-			return false;
+			return type.Is (Namespaces.CoreImage, "CIFilter") || IsCIFilter (type.Resolve ().BaseType);
```

Comment:
that sounds like a costly recursion in something that's called quite often
maybe it should be cached ? but let's get numbers first :) and the optimization can be done (if needed) in a separate PR